qcodes/utils/deprecate.py

```diff
@@ -46,7 +49,7 @@ def deprecate(
         alternative: The alternative function or class to put in use instead of
             the deprecated one.
     """
-    def actual_decorator(func: Callable) -> Callable:
+    def decorate_callable(func: Callable):
```

Comment:
shall we do smth like mentioned here https://mypy.readthedocs.io/en/latest/generics.html#declaring-decorators ? or is it too much for this PR?

```python
FuncType = Callable[..., Any]
F = TypeVar('F', bound=FuncType)

# A decorator that preserves the signature.
def my_decorator(func: F) -> F:
```

Reply:
This is nice! yes, I think we should!

Reply:
Does not work as expected :-( leaving this for now.